contrib/sarplus/azure-pipelines.yml

```diff
@@ -0,0 +1,93 @@
+# Python package
+# Create and test a Python package on multiple Python versions.
+# Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/python
+
+pr:
+  branches:
+    include:
+      - staging
+      - master
+  paths:
+    include:
+      - contrib/sarplus/*
+
+# no CI trigger
+trigger: none
+
+jobs:
+- job: 'Test'
+  pool:
+    vmImage: 'Ubuntu 16.04'
+  strategy:
+    matrix:
+      Python35-Spark2.3:
+        python.version: '3.5'
+        spark.version: '2.3.0'
+      Python36-Spark2.3:
+        python.version: '3.6'
+        spark.version: '2.3.0'
+      Python35-Spark2.4.1:
+        python.version: '3.5'
+        spark.version: '2.4.1'
+      Python36-Spark2.4.1:
+        python.version: '3.6'
+        spark.version: '2.4.1'
+      Python36-Spark2.4.3:
+        python.version: '3.6'
+        spark.version: '2.4.3'
+      Python37-Spark2.4.3:
+        python.version: '3.7'
+        spark.version: '2.4.3'
+    maxParallel: 4
+
+  steps:
+  - task: ComponentGovernanceComponentDetection@0
+    inputs:
+      scanType: 'Register'
+      verbosity: 'Verbose'
+      alertWarningLevel: 'High'
+      sourceScanPath: contrib/sarplus
+
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '$(python.version)'
+      architecture: 'x64'
+
+  # pyarrow version: https://issues.apache.org/jira/projects/SPARK/issues/SPARK-29367
+  - script: python -m pip install --upgrade pip && pip install pyspark==$(spark.version) pytest pandas pybind11 pyarrow==0.14.1 sklearn

```

Comment:
Do we want to reuse the env setup with the repo, like what we do for other testing?

Reply:
are you referring to scripts/tests/ci/env-setup.yml?

Reply:
Something like that - `scripts/testes/ci/env-setup.yaml` is used for AML based testing. We put something into each of the yaml files on the DSVM based testing. Not sure reusing of them will be helpful...